### ISSUE TITLE
Steps to install a system with external MQTT are not documented, and you cannot configure without a local MQTT broker

### ISSUE THREAD

**Issue by fake-name** [rhasspy/rhasspy#243]
I'm installing rhasspy on a fresh raspberry pi via dpkg.

I have the deb installed, but all configuration seems to be done via the web interface, and you cannot start the web interface without a MQTT broker, and the process for how to configure rhasspy to not just default to the local broker is undocumented.

Looking in `/usr/lib/rhasspy/rhasspy-profile/rhasspyprofile/profiles/en/profile.json`, there are no keys for anything related to `mqtt`. My local home directory `/home/pi/.config/rhasspy/profiles/en/profile.json` does not exist, and there are no entries for anything containing `rhasspy` under `/etc/` (which is where configuration files for daemons like this should really go).

Googling lead me to a random forum post (https://community.rhasspy.org/t/rhasspy-as-service-with-debian-installation/1278/9 ) with a json snippet that looked promising. Blindly sticking that in the root of the `/usr/lib/rhasspy/rhasspy-profile/rhasspyprofile/profiles/en/profile.json` json file (and fixing the fucking disqus smart quotes. Holy shit fuck why does discus alter code blocks and insert smart quotes?) now lets rhasspy at least start so I can configure stuff.

I think, if the default config file (`/usr/lib/rhasspy/rhasspy-profile/rhasspyprofile/profiles/en/profile.json`) at least included a stub `mqtt` section that specifies the localhost config, that would be adequate here.

Really, the complete reliance on the web UI for configuring things isn't.... *great*, but it only works if it doesn't depend on *any* external agents in order to start.

Also, maybe put the config files in `/etc/`, where system wide config files should go?

**Comment by siparker** [comment]
I would also like to see this config documented fully as my Mqtt installation is already running on a separate machine and used by the rest of my smart devices.

**Comment by donburch888** [comment]
I am writing a tutorial for a minimal satellite install with RasPi Zero WH and reSpeaker 2-mic HAT.  Given that RasPi Zero is comparatively slow, I would like to eliminate all the unnecessary services (like Docker).  

I have got two working configurations, but I did get to it via the Internal MQTT option ... and I notice that mosquitto service is still being started when the machine boots, despite it no longer being used. 

Reading your comments, I think I'll take a fresh SD card and try to configure the External MQTT version directly. I will try editing profile.json directly before starting Rhasspy, and see if that makes a change.  

Update: Yeah, can't be done ! Looks like installing mosquitto is the first thing Rhasspy install does, and the service stays enabled after Rhasspy is configured to External MQTT.  

Is localhost: mosquitto still used internally by Rhasspy (say, by the Dialogue Manager) ?

**Comment by Romkabouter** [comment]
When I set the MQTT to external, this is visible in the profile.json

```
    "mqtt": {
        "enabled": "true",
        "host": "192.168.43.133",
        "password": "rhasspy",
        "site_id": "base",
        "username": "rhasspy"
    },
```

The docs are indeed a bit lacking on this point
If you have found good steps to include this in the document, maybe you can contribute a bit and add them?

**Comment by fake-name** [comment]
> When I set the MQTT to external, this is visible in the `profile.json`

Yeah, but you can't *start the web interface which lets you set that* without a functional MQTT server. So if you don't set up MQTT on the localhost which it can connect to as part of the startup, you can't set that because the web interface is inaccessible.

I generally think having someone who basically has no idea what they're doing (e.g. me, who is trying to get a minimal system even marginally functional) writing the documentation is a *terrible* idea. I don't know how things are *supposed* to be done, so I'm possibly the worst person to be writing documentation here. Did I do anything in the original issue report correctly? **I have no idea!**. That's not the position you want the people writing documentation to be in.

Arguably, writing documentation is probably the task that takes the *deepest* knowledge about the system in question, as proper documentation encompasses the *why* of how things are done. It's probably the *worst* task to ask of new contributors.

**Comment by Romkabouter** [comment]
> Yeah, but you can't _start the web interface which lets you set that_ without a functional MQTT server. So if you don't set up MQTT on the localhost which it can connect to as part of the startup, you can't set that because the web interface is inaccessible.

You can edit the file directly, that is what I meant.

**Comment by siparker** [comment]
Just to query then @Romkabouter. if you setup rhasspy without mqtt you can then add an external mqtt directly to the profile.json which will allow you to have an external mqtt broker without having to setup the internal one initially?

is that correct? in principle at least?

**Comment by Romkabouter** [comment]
> is that correct? in principle at least?

Yes it is. If you put the json I posted earlier in the profile.json than Rhasspy will start with those settings.
Note that you should add it to the profile.json in the language you select.

**Comment by fake-name** [comment]
Hello! It's 3 years later, I just updated my old rhasspy install, and what do you know, this problem is still going on!

It also appears to have eaten my config somewhere in updating from buster to bullseye. so I'm back at "it doesn't work, and I don't even know where the config file is" again. Whoooo
